jsapp/scss/components/_kobo.map.scss

```diff
@@ -230,95 +275,104 @@ body .leaflet-control-layers,
 
-  &-4, &-4:after {
-    background: rgba(113, 230, 33, 0.4);
-  }
+$class-slug: map-marker !default;
 
-  &-5, &-5:after {
-    background: rgba(253, 35, 4, 0.4);
-  }
+$colorLength: length($color-collection);
 
-  &-6, &-6:after {
-    background: rgba(33, 231, 184, 0.4);
+@for $i from 1 through $colorLength {
+  .map-marker.#{$class-slug}-#{$i},
+  .map-marker.#{$class-slug}-#{$i + $colorLength},
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 2},
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 3},
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 4} {
+    background: nth($color-collection, $i);
   }
 
-  &-7, &-7:after {
-    background: rgba(55, 40, 210, 0.4);
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 2} {
+    background-image: linear-gradient(90deg, rgba(255, 255, 255, 0.4) 50%, transparent 50%), linear-gradient(rgba(255, 255, 255, 0.4) 50%, transparent 50%);
+    background-size: 6px 6px;
+    background-position: center center;
   }
 
-  &-8, &-8:after {
-    background: rgba(200, 80, 246, 0.4);
+  .map-marker.#{$class-slug}-#{$i + $colorLength} {
+    background-image: repeating-linear-gradient(45deg, transparent, transparent 3px, rgba(255, 255, 255, 0.3) 3px, rgba(255, 255, 255, 0.3) 6px);
   }
 
-  &-9, &-9:after {
-    background: rgba(208, 165, 50, 0.4);
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 3} {
+    background-image: repeating-linear-gradient(-45deg, transparent, transparent 3px, rgba(255, 255, 255, 0.3) 3px, rgba(255, 255, 255, 0.3) 6px);
   }
 
-  &-10, &-10:after {
-    background: rgba(232, 65, 14, 0.4);
+  .map-marker.#{$class-slug}-#{$i + $colorLength * 4} {
+    background-image: linear-gradient(90deg, rgba(0, 0, 0, 0.1) 50%, transparent 50%), linear-gradient(rgba(0, 0, 0, 0.1) 50%, transparent 50%);
+    background-size: 6px 6px;
   }
+}
 
-  &-11, &-11:after {
-    background: rgba(216, 150, 216, 0.4);
-  }
+// ClusterMarker overrides
 
-  &-12, &-12:after {
-    background: rgba(200, 233, 150, 0.4);
-  }
+.marker-cluster-small {
+  background-color: rgba(253, 187, 132, 0.4);
+}
 
-  &-13, &-13:after {
-    background: rgba(122, 187, 200, 0.4);
-  }
+.marker-cluster-small div {
+  background-color: rgba(253, 187, 132, 0.5);
```

Comment:
Wouldn't `background-color: inherit` be good enough? :)

Reply:
It wouldn't have the subtle opacity difference.